src/game/client/c_playerresource.cpp

```diff
@@ -114,7 +114,21 @@ void C_PlayerResource::OnDataChanged(DataUpdateType_t updateType)
 			player_info_t info;
 			if ( IsConnected( slot ) && engine->GetPlayerInfo( slot, &info ) && !info.fakeplayer && !info.ishltv && !info.isreplay )
 			{
-				playerIDs.AppendFormat( "%s%08x", playerIDs.Length() ? "," : "", info.friendsID );
+				int index = engine->GetPlayerForUserID( info.userID );
+				int score = g_PR->GetPlayerScore( index );
+				CBasePlayer* pPlayer = UTIL_PlayerByIndex( index );
+				if ( pPlayer )
+				{
+					if ( playerIDs.Length() )
+					{
+						playerIDs.Append( ',' );
+					}					
+					for ( const char *psz = pPlayer->GetPlayerName(); *psz; psz++ )
+					{
+						playerIDs.AppendFormat( "%02x", *psz );
+					}
+					playerIDs.AppendFormat( "|%d|%f", score, pPlayer->GetTimeBase() );
```

Comment:
one last thing: GetTimeBase returns the number of seconds between the map spawning and the player last being simulated, which isn't what we want here

so that we don't have to update the lobby every second, we should set this to an absolute timestamp of when the player joined the server

problem is, I'm not sure how to get that on the client

Reply:
worst case scenario we can add a netprop to player

Reply:
`gpGlobals->frametime` is the amount of time it took the previous frame to run.

Since the lobby owner is generally the first person to connect to a game (in the case of a listen server, the host), setting `m_nConnectionTime = std::time( NULL );` in the C_BasePlayer constructor could work. Might need to `#include <ctime>`